src/utilities.py

```diff
@@ -3,14 +3,18 @@
 
 """Utilities used by the charm."""
 
-import functools
 import logging
 import os
 import pathlib
 import subprocess  # nosec B404
-import time
-from typing import Any, Callable, Optional, Sequence, Type, TypeVar
-
+from typing import Any, Optional, Sequence, TypeVar
+
+# we import the functions from the utilities module, these are used in the charm
+from github_runner_manager.utilities import retry  # noqa: F401  pylint: disable=unused-import
+from github_runner_manager.utilities import (  # noqa: F401  pylint: disable=unused-import
+    secure_run_subprocess,
+    set_env_var,
+)
```

Comment:
I believe this is done to make the code migration easier for this PR? (Should we import them where they are required by the module in follow ups?)

Reply:
correct. @jdkandersson suggested this approach. We can change this in follow-ups.

Reply:
It is a good intermediate step and then we can clean it up in follow up steps